gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -468,6 +562,13 @@ private void handleContainerCompletion(ContainerStatus containerStatus) {
           containerStatus.getContainerId(), containerStatus.getDiagnostics()));
     }
 
+    if (this.releasedContainerSet.contains(containerStatus.getContainerId())) {
+      LOGGER.info("Container release requested, so not spawning a replacement for containerId {}",
+          containerStatus.getContainerId());
+      this.releasedContainerSet.remove(containerStatus.getContainerId());
```

Comment:
Why is this remove safe to perform. Are we assured that spawning a replacement for this containerId will not happen again?